React/Views/RCTMapManager.m

```diff
@@ -312,6 +312,19 @@ - (void)mapView:(RCTMap *)mapView annotationView:(MKAnnotationView *)view
   }
 }
 
+- (void)mapView:(RCTMap *)mapView didAddAnnotationViews:(NSArray *)views
+
+{   
+  for (MKAnnotationView* view in views) {
+    if ([view.annotation isKindOfClass:[RCTMapAnnotation class]]) {
+      RCTMapAnnotation *annotation = (RCTMapAnnotation *)view.annotation;
+      if (annotation.selectAnnotation) {
```

Comment:
Assuming that an unselect occurs by tapping outside the callout the [didDeselectAnnotationView](https://developer.apple.com/library/ios/documentation/MapKit/Reference/MKMapViewDelegate_Protocol/#//apple_ref/occ/intfm/MKMapViewDelegate/mapView:didDeselectAnnotationView:) is called which updates the annotation.